src/simple_key_provider.rs

```diff
@@ -2,7 +2,7 @@
 
 use aes_gcm::aead::{Aead, Payload};
 use aes_gcm::aes::cipher::consts::U16;
-use aes_gcm::aes::Aes128; // Or Aes256
+use aes_gcm::aes::Aes128;
```

Comment:
You shouldn't need to import this if you've already used `Aes128Gcm` etc.

But looking at the whole file, it appears that the macro still refers to `Aes128` directly where as it should be using `$name`.

Reply:
Specifically:

```rust
let key = Key::<Aes128>::from_slice(&self.kek);
let cipher = AesGcm::<Aes128, U16>::new(key);
```

Should be:

```rust
let key = Key::<$name>::from_slice(&self.kek);
let cipher = AesGcm::<$name, $key_size>::new(key);
```

Reply:
I've just realised that you're doing this for the KEK and not the data-key which confused me. 2 suggestions to make it clearer:

* Consider initializing the `cipher` in the init function to separate it from the trait impl
* OR use the same sized KEK as Datakey

Reply:
I was confused about this as well. For some reason, we use `AesGcm<Aes128, U16>`, instead of `Aes128Gcm = AesGcm<Aes128, U12>` for encrypting the data key, despite the fact we use `Aes128Gcm` for the actual envelope encryption.

I thought there must have been some historical reason why it is done differently, so I did not touch it.

Do you think data key encryption should use the same cipher as the envelope encryption?

Reply:
I'm moving `cipher` initialization into the init function for now.

Reply:
Yeah I think for the simple key provider this is fine. Much clearer now!